Common/DeviceSchema/DeviceSchemaHelper.cs

```diff
@@ -42,6 +42,32 @@ public static dynamic GetDeviceProperties(dynamic device)
         }
 
         /// <summary>
+        /// Gets a DeviceProperties instance from a Device.
+        /// </summary>
+        /// <param name="device">
+        /// The Device from which to extract a DeviceProperties instance.
+        /// </param>
+        /// <returns>
+        /// A DeviceProperties instance, extracted from <paramref name="device"/>.
+        /// </returns>
+        public static dynamic GetDevicePropertiesND(DeviceND device)
+        {
+            if (device == null)
+            {
+                throw new ArgumentNullException("device");
+            }
+
+            var props = device.DeviceProperties;
+
+            if (props == null)
+            {
+                throw new DeviceRequiredPropertyNotFoundException("'DeviceProperties' property is missing");
+            }
+
```

Comment:
this can go inline so we can remove the schema helper when we're done refactoring